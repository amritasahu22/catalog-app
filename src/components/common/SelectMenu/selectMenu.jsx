import React from "react";
import PropTypes from "prop-types";

const SelectMenu = (props) => {
	const { options, selectedOption, onOptionSelect } = props;

	return (
		<select
			className="custom-select clickable"
			value={selectedOption}
			onChange={(e) => onOptionSelect(e.target.value)}
		>
			{options.map((option) => (
				<option key={option.id} value={option.value}>
					{option.name}
				</option>
			))}
		</select>
	);
};

SelectMenu.propTypes = {
	options: PropTypes.array.isRequired,
	selectedOption: PropTypes.string,
	onOptionSelect: PropTypes.func.isRequired,
};

export default SelectMenu;
