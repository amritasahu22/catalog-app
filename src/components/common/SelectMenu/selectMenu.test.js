import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import SelectMenu from "./selectMenu";

describe("SelectMenu", () => {
	const options = [
		{ id: 1, name: "Popularity", value: "popularity" },
		{ id: 2, name: "New Arrivals", value: "new" },
	];

	test("should render properly with props", () => {
		const component = shallow(
			<SelectMenu
				options={options}
				selectedOption={options[2]}
				onOptionSelect={() => {}}
			/>
		);
		expect(toJson(component)).toMatchSnapshot();
	});
});
