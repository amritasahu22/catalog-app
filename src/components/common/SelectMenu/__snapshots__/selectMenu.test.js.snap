// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectMenu should render properly with props 1`] = `
<select
  className="custom-select clickable"
  onChange={[Function]}
>
  <option
    key="1"
    value="popularity"
  >
    Popularity
  </option>
  <option
    key="2"
    value="new"
  >
    New Arrivals
  </option>
</select>
`;
